components/feature/downloads/src/test/java/mozilla/components/feature/downloads/AbstractFetchDownloadServiceTest.kt

```diff
@@ -1135,6 +1136,62 @@ class AbstractFetchDownloadServiceTest {
         verify(service).addToDownloadSystemDatabaseCompat(any())
     }
 
+    @Test
+    fun `WHEN we download on devices with version higher than Q which use legacy external storage THEN we use legacy file stream`() {
+        val service = spy(object : AbstractFetchDownloadService() {
+            override val httpClient = client
+            override val store = browserStore
+        })
+        val uniqueFile: DownloadState = mock()
+        val qSdkVersion = 29
```

Comment:
Nit: We can change the SDK via [@Config(sdk )](http://robolectric.org/configuring/) :)